Code/GraphMol/MolDraw2D/DrawMol.cpp

```diff
@@ -819,8 +820,7 @@ void DrawMol::calculateScale() {
       height_ = drawHeight_;
       legendHeight_ = 0;
     } else {
-      height_ = drawHeight_ / (1.0 - drawOptions_.legendFraction);
-      drawHeight_ = height_ - legendHeight_;
+      height_ = drawHeight_;
```

Comment:
This is in both sides of the condition, so move it above, and remove the empty else?
Also maybe do the if (height_ < 0) test as if(flexiCanvasY_) and likewise for (width_ < 0) throughout this function?  Now we have the bools, it makes it a bit clearer what is going on.